main.py

```diff
@@ -143,17 +245,22 @@ def heartbeat() -> None:
         # cleanup, we can just skip GC and exit immediately.
         os._exit(0)
 
+# This has to be a global since we need to access the permuters. 
+# multiprocessing fork()'s the main thread and it works out that the permuters
+# are unmodified because we are simply passing in pre-calculated seeds from the
+# main thread. In theory, we should be able to pass in the context as an argument
+# but haven't been able to get pass various exceptions (probably related to pickling).
```

Comment:
sounds odd, given that Candidates apparently can be sent through task queues... what exception do you get?